File: app/src/main/java/org/dhis2/data/user/UserModule.java
package org.dhis2.data.user;


import org.dhis2.data.dagger.PerUser;
import org.hisp.dhis.android.core.D2;

import dagger.Module;
import dagger.Provides;

@Module
@PerUser
public class UserModule {

    @Provides
    @PerUser
    UserRepository userRepository(D2 d2) {
        return new UserRepositoryImpl(d2);
    }

}

File: app/src/main/java/org/dhis2/data/user/UserModule.kt
package org.dhis2.data.user

import dagger.Module
import dagger.Provides
import org.dhis2.commons.di.dagger.PerUser
import org.hisp.dhis.android.core.D2

@Module
class UserModule {
    @Provides
    @PerUser
    fun userRepository(d2: D2?): UserRepository {
        return UserRepositoryImpl(d2!!)
    }
}
